### PR TITLE
Handle No Projects Correctly - fixes #138

### LINKED ISSUE #138: No Billing Error
We have handling for helping the user setup billing, but unfortunately it doesn't work in all scenarios.  In a new account I'm getting:
```
demo@cloudshell:~$ cloudshell_open --repo_url "https://github.com/jamesward-demo/hello-php.git" --page "editor" --git_branch "master"
[ ✓ ] Cloned git repository https://github.com/jamesward-demo/hello-php.git.
[ ✓ ] Queried list of your GCP projects
[ ? ] Would you like to use existing GCP project  to deploy this app? Yes
Error: error determining if billing account is linked: exit status 1. output=
ERROR: (gcloud.beta.billing.projects.describe) could not parse resource []
```

### PR DESCRIPTION
**Pull request by jamesward**
strings.Split on an empty string returns a slice of length 1 containing an empty string.  We need to detect this and return a new slice of length 0.

### CODE REVIEW

File: cmd/cloudshell_open/project.go [hunk 1]
```diff
@@ -43,8 +43,12 @@ func listProjects() ([]string, error) {
 	}
 	b = bytes.TrimSpace(b)
 	p := strings.Split(string(b), "\n")
-	sort.Strings(p)
-	return p, err
+	if len(p) == 1 && p[0] == "" {
+		return []string{}, err
```
Comment (ahmetb) [inline cmd/cloudshell_open/project.go]: you can do `return nil, err`, nil is a valid value for slices.

File: cmd/cloudshell_open/project.go [hunk 2]
```diff
@@ -43,8 +43,12 @@ func listProjects() ([]string, error) {
...
+	} else {
```
Comment (ahmetb) [inline cmd/cloudshell_open/project.go]: no need for `else{` because you return in previous block.

File: cmd/cloudshell_open/project.go [hunk 3]
```diff
@@ -43,8 +43,12 @@ func listProjects() ([]string, error) {
 	}
 	b = bytes.TrimSpace(b)
 	p := strings.Split(string(b), "\n")
-	sort.Strings(p)
-	return p, err
+	if len(p) == 1 && p[0] == "" {
```
Comment (ahmetb) [inline cmd/cloudshell_open/project.go]: I think the cleaner course of action would be this:


```
b := string(bytes.TrimSpace(b))
if b == "" {
   return nil, nil
}

// split
// sort
// return
```

File: cmd/cloudshell_open/project.go [hunk 4]
```diff
@@ -43,8 +43,12 @@ func listProjects() ([]string, error) {
...
+		return []string{}, err
+	} else {
+		sort.Strings(p)
+		return p, err
```
Comment (ahmetb) [inline cmd/cloudshell_open/project.go]: both `return x, err` should be `return x, nil`.
not sure why `err` is down here..

File: cmd/cloudshell_open/project.go (hunk 2 above)
Comment (jamesward) [reply cmd/cloudshell_open/project.go]: Hmm...  Don't I need to return if the if didn't match?

Comment (ahmetb) [reply cmd/cloudshell_open/project.go]: I meant this, if you do

```
if cond {
   return
} else {
   logic
   return
}
```

several Go linting tools would tell you `else is superfluous` as it's equivalent to

```
if cond {
   return
}

logic
return
```

which is more readable.

**Comment by ahmetb** [comment]
Consider having `fixes #xxx` in the commit/PR message (not title) so github can auto-close :)

File: cmd/cloudshell_open/project.go (hunk 2 above)
Comment (jamesward) [reply cmd/cloudshell_open/project.go]: Ah. I see.  I'm used to expression-oriented ifs :)

**Comment by jamesward** [comment]
> Consider having `fixes #xxx` in the commit/PR message (not title) so github can auto-close :)

It just needs to be in the first commit (which it was).  The PR picked the title from the first commit.

I've cleaned this up to your suggestion.

**Comment by ahmetb** [comment]
Awesome, thanks! 
The CI will take ~forever. Feel free to merge yourself.

**Comment by jamesward** [comment]
Thanks for your Go styling help :)

**Pull request merged**